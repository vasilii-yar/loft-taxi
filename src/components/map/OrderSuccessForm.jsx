import React from 'react';
import Paper from "@material-ui/core/Paper";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import {connect} from "react-redux";
import {clearRoute, makeOrderSuccess} from "../../redux/modules/route/routeActions";
import "./OrderSuccessForm.css";

const OrderSuccessForm = (props) => {
    const handleSubmit = (e) => {
        e.preventDefault();
        props.makeOrderSuccess(false);
        props.clearRoute();
    }

    return (
        <Paper className="order-success-form-layout">
            <form onSubmit={handleSubmit}>
                <Grid container direction="column" spacing={2}>
                    <Grid item xs={12}>
                        <Typography variant="h4">
                            Заказ размещён
                        </Typography>
                    </Grid>
                    <Grid item xs={12}>
                        <Typography>
                            Ваше такси уже едет к вам. Прибудет приблизительно через 10 минут.
                        </Typography>
                    </Grid>
                    <Grid item xs={12}>
                        <Button variant="contained" color="primary" type="submit" fullWidth>
                            Сделать новый заказ
                        </Button>
                    </Grid>
                </Grid>
            </form>

        </Paper>
    )
}

export default connect(
    null,
    {makeOrderSuccess, clearRoute}
)(OrderSuccessForm);