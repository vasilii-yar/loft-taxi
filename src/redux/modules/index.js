import authReducer from "./auth/authReducer";
import profileReducer from "./profile/profileReducer";
import {combineReducers} from "redux";


export default combineReducers({
    auth: authReducer,
    profile: profileReducer
});