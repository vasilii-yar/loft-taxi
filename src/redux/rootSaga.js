import {all} from "@redux-saga/core/effects";
import {authSaga} from "./modules/auth/authSaga";
import {profileSaga} from "./modules/profile/profileSaga";
import {addressSaga} from "./modules/address/addressSaga";
import {routeSaga} from "./modules/route/routeSaga";

export default function* rootSaga() {
    yield all([
        authSaga(),
        profileSaga(),
        addressSaga(),
        routeSaga()
    ])
}