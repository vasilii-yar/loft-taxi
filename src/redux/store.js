import {applyMiddleware, createStore} from "redux";
import mainReducer from "./modules"
import {authMiddleware} from "./middleware/authMiddleware";
import {profileMiddleware} from "./middleware/profileMiddleware";

export const store = createStore(mainReducer, applyMiddleware(authMiddleware, profileMiddleware));