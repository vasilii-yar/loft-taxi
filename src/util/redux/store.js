import {applyMiddleware, createStore} from "redux";
import mainReducer from "../../reducers"
import {authMiddleware} from "../../middleware/authMiddleware";

export const store = createStore(mainReducer, applyMiddleware(authMiddleware));