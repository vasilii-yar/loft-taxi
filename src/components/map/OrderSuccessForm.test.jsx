import React from 'react';
import {render} from "@testing-library/react";
import {BrowserRouter} from "react-router-dom";
import {Provider} from "react-redux";
import {store} from "../../redux/store";
import renderer from "react-test-renderer";
import OrderSuccessForm from "./OrderSuccessForm";

describe("OrderSuccessForm", () => {
    it("renders correctly", () => {
        const {container} = render(
            <BrowserRouter>
                <Provider store={store}>
                    <OrderSuccessForm/>
                </Provider>
            </BrowserRouter>
        );
        expect(container.innerHTML).toMatch("Заказ размещён");
    })

    it("renders correctly check by snapshot", () => {
        const tree = renderer.create(
            <BrowserRouter>
                <Provider store={store}>
                    <OrderSuccessForm/>
                </Provider>
            </BrowserRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})