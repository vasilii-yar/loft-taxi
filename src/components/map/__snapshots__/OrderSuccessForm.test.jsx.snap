// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderSuccessForm renders correctly check by snapshot 1`] = `
<div
  className="MuiPaper-root order-success-form-layout MuiPaper-elevation1 MuiPaper-rounded"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4"
        >
          Заказ размещён
        </h4>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Ваше такси уже едет к вам. Прибудет приблизительно через 10 минут.
        </p>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Сделать новый заказ
          </span>
        </button>
      </div>
    </div>
  </form>
</div>
`;
